ui/ceval/src/stockfishProtocol.js

```diff
@@ -36,48 +42,52 @@ module.exports = function(worker, opts) {
     }
     if (!work) return;
 
-    if (text.indexOf('currmovenumber') !== -1) return;
-    var matches = text.match(/depth (\d+) .*multipv (\d+) .*score (cp|mate) ([-\d]+) .*nps (\d+) .*pv (.+)/);
-    if (!matches) {
-      emit();
-      return;
-    }
+    var matches = text.match(EVAL_REGEX);
+    if (!matches) return;
+
+    var depth = parseInt(matches[1]),
+        multiPv = parseInt(matches[2]),
+        isMate = matches[3] === 'mate',
+        eval = parseFloat(matches[4]),
+        evalType = matches[5],
+        nodes = parseInt(matches[6]),
+        nps = parseInt(matches[7]),
+        elapsedMs = parseInt(matches[8]),
+        pv = matches[9];
 
-    var depth = parseInt(matches[1]);
     if (depth < opts.minDepth) return;
-    var multiPv = parseInt(matches[2]);
-    var cp, mate;
-    if (matches[3] === 'cp') cp = parseFloat(matches[4]);
-    else mate = parseFloat(matches[4]);
-    if (work.ply % 2 === 1) {
-      if (matches[3] === 'cp') cp = -cp;
-      else mate = -mate;
+    if (work.ply % 2 === 1) eval = -eval;
+
+    if (evalType) return; // Ignore lowerbound and upperbound for now.
+
+    var pvData = {
+      best: pv.split(' ', 2)[0],
+      pv: pv,
+      cp: isMate ? undefined : eval,
+      mate: isMate ? eval : undefined,
     }
 
+    // Ignore output from earlier depths.
+    if (state && depth < state.eval.depth) return;
+
     if (multiPv === 1) {
-      emit();
```

Comment:
I was guessing it was here because `if (multiPv === opts.multiPv)` is never true, so to work around an issue with emits.